src/ansys/fluent/core/workflow.py

```diff
@@ -902,10 +902,16 @@ def __setattr__(self, attr, value):
             self.set_state({attr: value})
 
     def __dir__(self):
-        arg_list = []
-        for arg in self():
-            arg_list.append(camel_to_snake_case(arg))
-        return sorted(set(list(self.__dict__.keys()) + dir(type(self)) + arg_list))
+        arg_state = self.get_state()
+        arg_list = (
+            [arg for arg in arg_state if isinstance(arg_state, dict)]
+            if self() is not None
```

Comment:
Is `self()` same as `arg_state`? Instead of calling get_state again can check `arg_state is not None`

Reply:
Good catch, Thanks.
Done.

Reply:
`arg_state = list(args_state) if isinstance(arg_state, dict) else []`